frontend/src/component/segments/SegmentTable.tsx

```diff
@@ -39,17 +41,22 @@ export const SegmentTable = () => {
     });
 
     const data = useMemo(() => {
-        return (
-            segments ??
-            Array(5).fill({
+        if (!segments) {
+            return Array(5).fill({
                 name: 'Segment name',
                 description: 'Segment descripton',
                 createdAt: new Date().toISOString(),
                 createdBy: 'user',
                 projectId: 'Project',
-            })
-        );
-    }, [segments]);
+            });
+        }
+
+        if (projectId) {
+            return segments.filter(({ project }) => project === projectId);
+        }
```

Comment:
I believe you should move the segments filter before line 44 where you check `if (!segments)`, because you could have segments at the beginning but maybe not after filtering by project

Reply:
I think I understand what you mean but I believe the logic is a bit different here than what you assumed in your comment. 
If you check line 44 again, we are checking for truthy `segments`, not their length. The purpose of the first check on line 44 is to fill segments with placeholder information for loading purposes (in case `segments` is still `undefined`, let's fill them with mock info to show as loading skeletons or what not). 
From that point forward, we are totally fine with having `[]` for segments here, and the table should handle that by displaying an appropriate message. In fact, we could not even filter them on line 55 if they were still `undefined`.

Reply:
Gotcha! I thought it was something different. I also see it was there before :+1: